buy_together_app/models.py

```diff
@@ -1,3 +1,55 @@
-# from django.db import models
+from django.db import models
+from django.core.validators import MinLengthValidator
+from django.core.exceptions import ValidationError
 
-# Create your models here.
+
+class Product(models.Model):
+    qr_code = models.CharField(max_length=32, validators=[MinLengthValidator(32)], primary_key=True)
+    product_name = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    description = models.TextField(validators=[MinLengthValidator(1)])
+
+    @staticmethod
+    def filter_qr(qr):
+        if qr:
+            return Product.objects.filter(qr_code=qr)
+        return Product.objects.all()
+
+    @staticmethod
+    def filter_name(nm):
+        if nm:
+            return Product.objects.filter(product_name=nm)
+        return Product.objects.all()
+
+    @staticmethod
+    def filter_description(desc):
+        if desc:
+            return Product.objects.filter(description=desc)
+        return Product.objects.all()
+
+    @staticmethod
+    def get_all_products():
+        return Product.objects.all()
+
+    @staticmethod
+    def save_product(prd):
+        try:
+            prd.save()
+            Product.full_clean(prd)
+            return prd
+        except ValidationError:
+            if len(prd.product_name) == 0:
+                prd.delete()
+                raise ValidationError("product name cant be zero")
+            elif len(prd.description) == 0:
+                prd.delete()
+                raise ValidationError("product description cant be zero")
+            elif len(prd.qr_code) != 32:
+                prd.delete()
+                raise ValidationError("product qr cant be other than 32")
```

Comment:
I still don't understand why do we need to replicate the validation code. What happens if you'll just let it raise the exception?

Maybe I'm missing something. @Omeramsc will you be able to take a look?